src/main/java/com/shop/service/UserServiceImpl.java

```diff
@@ -10,49 +10,48 @@
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.HashSet;
+import java.util.Optional;
 import java.util.Set;
 
 @Service
 public class UserServiceImpl implements UserService {
 
-    @Autowired
     private UserRepository userRepository;
+    private RoleRepository roleRepository;
+
     @Autowired
-    private RoleService roleService;
-    @Autowired
-    private BCryptPasswordEncoder passwordEncoder;
+    public UserServiceImpl(UserRepository userRepository, RoleRepository roleRepository) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+    }
 
     @Override
     public User register(User user) {
+        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+
         user.setEnabled(true);
-        roleService.addUserRole(user);
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
-        if(userRepository.findByUsername(user.getUsername()) != null || userRepository.findByEmail(user.getEmail()) != null)
-            return null;
+        user.setRegistrationDate(Calendar.getInstance());
+        addUserRole(user);
+        user.setPassword(encoder.encode(user.getPassword()));
         return userRepository.save(user);
     }
 
     @Override
-    public User addAdminUser() {
-        User admin = userRepository.findByUsername("admin");
-        if (admin == null) {
-            admin = new User(Calendar.getInstance(), "admin", passwordEncoder.encode("admin"), "admin@admin.com");
-            admin.setEnabled(true);
+    public void addAdminUser() {
+        Optional<User> admin = userRepository.findByUsername("admin");
+        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+
+        if (admin.isPresent()) {
+            User user = new User(Calendar.getInstance(), "admin", encoder.encode("admin"), "admin@admin.com");
```

Comment:
Shouldn't there be a negation in the if statement? Because right now if there is an admin user in DB u are creating new admin user, and if there is no admin user u are skipping whole if block